### PR TITLE
disabling CSP for development environment only.

### PR DESCRIPTION
**Pull request by Anxhela21**
Not related to specific issue but to development environment and accessing the graphQL api server interface.  

### Description of changes
- Added conditional statement to check if dev environment in use and, if so, to disable CSP (Content Security Policy) to avoid running into CSP violations when trying to access the graphQL interface.

### CODE REVIEW

File: src/v2/index.js
Comment (jlpadilla) [inline src/v2/index.js:104]: This should do the same with simpler code.
```suggestion
  graphQLServer.use('*', helmet({
    frameguard: false,
    noSniff: false,
    xssFilter: false,
    contentSecurityPolicy: process.env.NODE_ENV === 'production' // False for development
  }), noCache(), requestLogger, cookieParser());
```

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold
@Anxhela21 Since we are at the end of the release, let's be extra cautious and test the PR image in a cluster before merging this change.

**Comment by Anxhela21** [comment]
Tested with image `quay.io/open-cluster-management/search-api:2.3.0-PR123-c168d0d31dff3be0348514116d19d5fe28bebdb5` with no issues.

**Pull request merged**
